Rakefile

```diff
@@ -1,5 +1,154 @@
 require "bundler"
 Bundler.require
+require 'smarter_csv'
 
 require "sinatra/activerecord/rake"
 
+require File.expand_path('../config/environment', __FILE__)
+
+
+namespace :import do
+  desc "Import station.csv to cities table"
+  task :cities do
+   create_cities
+  end
+
+  desc "Import station.csv to table"
+  task :stations do
+   create_stations
+  end
+
+  desc "Import station.csv to subscription_types table"
+  task :subscription_types do
+   create_subscription_types
+  end
+
+  desc "Import weather.csv to table"
+  task :weather_conditions do
+    create_weather_conditions
+  end
+
+  desc "Import trip.csv to table"
+  task :trips do
+    create_trips
+  end
+
+end
+
+def create_cities
+  cities = SmarterCSV.process('db/csv/station.csv').map do |row|
+    row[:city]
+  end.uniq
+
+  cities.each do |city|
+    City.create(name: city)
+  end
+
+  puts "Imported Cities to Table."
+end
+
+def create_stations
+  def format_date(date)
+    date = date.split("/").reverse
+    date[1], date[2] = date[2], date[1]
+    date = date.join("/")
+    date
+  end
+
+  SmarterCSV.process('db/csv/station.csv').each do |row|
+    row[:installation_date] = format_date(row[:installation_date])
+    row[:city] = City.find_by(name: row[:city])
+    Station.create(row)
+  end
+
+  puts "Imported Stations to Table."
+end
+
+def create_subscription_types
+  subscription_types = SmarterCSV.process('db/csv/trip.csv').map do |row|
+    row[:subscription_type]
+  end.uniq
+
+  subscription_types.each do |subscription_type|
+    SubscriptionType.create(subscription_type: subscription_type)
+  end
+
+  puts "Imported Subscription Types to Table."
+end
+
+def create_weather_conditions
+  def format_date(date)
+    date = date.split("/").reverse
+    date[1], date[2] = date[2], date[1]
+    date = date.join("/")
+    date
+  end
+
+  SmarterCSV.process('db/csv/weather.csv',
+                    key_mapping: {date: :date,
+                                  max_temperature_f: :max_temperature_f,
+                                  mean_temperature_f: :mean_temperature_f,
+                                  min_temperature_f: :min_temperature_f,
+                                  mean_humidity: :mean_humidity,
+                                  mean_visibility_miles: :mean_visibility_miles,
+                                  mean_wind_speed_mph: :mean_wind_speed_mph,
+                                  precipitation_inches: :precipitation_inches,
+                                  zip_code: :zip_code},
+                    remove_unmapped_keys: true).each do |row|
+    row[:date] = format_date(row[:date])
+    row[:precipitation_inches] = row[:precipitation_inches].to_f
+    WeatherCondition.create(row)
+  end
+
+  puts "Imported Weather Conditions to Table."
+end
+
+def create_trips
+  def format_date(date)
```

Comment:
this method is defined multiple times. how can we DRY up this code and only write this method once?